pika/heartbeat.py

```diff
@@ -8,38 +8,37 @@
 
 
 class HeartbeatChecker(object):
-    """Checks to make sure that our heartbeat is received at the expected
-    intervals.
+    """Sends heartbeats to the broker and checks to make sure that the broker's
+    heartbeat is received before the expected timeout expires.
 
     """
-    DEFAULT_INTERVAL = 60
-    MAX_IDLE_COUNT = 2
-
+    _MAX_IDLE_COUNT = 2
     _STALE_CONNECTION = "Too Many Missed Heartbeats, No reply in %i seconds"
 
-    def __init__(self, connection, interval=DEFAULT_INTERVAL, idle_count=MAX_IDLE_COUNT):
-        """Create a heartbeat on connection sending a heartbeat frame every
-        interval seconds.
+    def __init__(self, connection, interval):
+        """Create a heartbeat on the connection that sends two heartbeat frames
+        within the specified timeout window. Also checks to ensure heartbeats
+        are received from the broker.
 
         :param pika.connection.Connection: Connection object
         :param int interval: Heartbeat check interval. Note: heartbeats will
-                             be sent at interval / 2 frequency.
-        :param int idle_count: The number of heartbeat intervals without data
-                               received that will close the current connection.
+                             actually be sent at interval / 2 frequency, and
+                             heartbeat checks made at (interval * 3) / 2. Default
+                             interval is 60 seconds so Pika will send heartbeats
+                             at 30 second intervals and will check every 90 seconds.
+                             If no heartbeat is received from the broker, nor data
+                             activity on the connection after 180 seconds the
+                             connection will be assumed dead and closed.
 
         """
         self._connection = connection
 
-        # Note: see the following document:
+        # Note: see the following documents:
         # https://www.rabbitmq.com/heartbeats.html#heartbeats-timeout
-        self._interval = float(interval / 2)
-
-        # Note: even though we're sending heartbeats in half the specified
-        # interval, the broker will be sending them to us at the specified
-        # interval. This means we'll be checking for an idle connection
-        # twice as many times as the broker will send heartbeats to us,
-        # so we need to double the max idle count here
-        self._max_idle_count = idle_count * 2
+        # https://github.com/pika/pika/pull/1072
+        # https://groups.google.com/d/topic/rabbitmq-users/Fmfeqe5ocTY/discussion
+        self._send_interval = float(interval) / 2
+        self._check_interval = (float(interval) * 3) / 2
```

Comment:
Not sure why I originally suggested the `* 3 / 2` math. My bad. It would be more readable as `timeout * 1.5` and floating point math would be happy on both 2.x and 3.x too.

Reply:
Alright, and if there's no activity when a check happens, close the connection.